R/zchunk_LA118.hydro.R

```diff
@@ -29,41 +29,170 @@ module_energy_LA118.hydro_DISABLED <- function(command, ...) {
+    # Interpolate any between years, and use the final year as a proxy for years thereafter
+    # Drop 2020 because it is lower than 2010 in many regions
+    L118.out_EJ_RG3_elec_hydro_fby %>%
+      bind_rows(
+        mutate(year = as.integer(year),
+          gather(A18.hydro_output, year, value, -region_GCAM3))) %>%
+      filter(year != 2020) -> # Dropping 2020, as described above
+      L118.out_EJ_RG3_elec_hydro_Y_with_values
+
+    L118.out_EJ_RG3_elec_hydro_fby %>%
+      select(region_GCAM3) %>%
+      repeat_add_columns(tibble::tibble(year = c(max(HISTORICAL_YEARS), FUTURE_YEARS))) %>% # Years include historical max and future
+      left_join(L118.out_EJ_RG3_elec_hydro_Y_with_values, by = c("region_GCAM3", "year")) %>%
+      group_by(region_GCAM3) %>%
+      mutate(value = approx_fun(year, value, rule = 2)) %>% # Interpolation step
+      ungroup() ->
+      L118.out_EJ_RG3_elec_hydro_Y_interp
+
+    # Calculate the growth from the base year in each region_GCAM3. This will be partitioned to countries according to potential
+    L118.out_EJ_RG3_elec_hydro_Y_interp %>%
+      filter(year != max(HISTORICAL_YEARS)) %>% # Deleting historical max year to be used in another column
+      left_join(
+        select(filter(L118.out_EJ_RG3_elec_hydro_Y_interp, year == max(HISTORICAL_YEARS)), -year), # Adding historical max year column so as to add base value to future years
+        by = "region_GCAM3") %>%
+      mutate(value_growth = value.x - value.y) %>%
+      select(-value.x, -value.y) ->
+      L118.growth_EJ_RG3_elec_hydro_Y
+
+    # Calculate the share of the growth potential by each country within its GCAM 3.0 region
+    Hydropower_potential %>%
+      group_by(region_GCAM3) %>%
+      summarise(Growth_potential_EJ_R = sum(Growth_potential_EJ)) %>%
+      filter(!is.na(region_GCAM3)) -> # Getting rid of the NA caused by Kosova not having a RG3 name
+      Hydropower_potential_RG3 # Represents total pie of each country
+
+    Hydropower_potential %>%
+      left_join(Hydropower_potential_RG3, by = "region_GCAM3") %>%
+      mutate(share = Growth_potential_EJ / Growth_potential_EJ_R) %>% # Calculating share
+    # Add future years to table of base-year hydropower generation as base-year output plus region_GCAM3 growth times country-wise share
+      select(iso, share) %>% # Need only share of each iso
+      group_by(iso) %>%
+      summarise(share = sum(share)) -> # Summing up Serbia...was listed twice
+      Hydropower_potential
+
+      # Adding future years
+    L118.out_EJ_ctry_elec_hydro_fby %>%
+      left_join(Hydropower_potential, by = "iso") %>% # Adding share column
+      left_join(L118.growth_EJ_RG3_elec_hydro_Y, by = "region_GCAM3") %>% # Adding RG3 growth
+      mutate(value_future = value_base + share * value_growth) %>% # Base-year output plus RG3 growth times country-wise share
+      select(-share, -value_growth) %>%
+      spread(year.x, value_base) %>%
+      spread(year.y, value_future) %>%
```

Comment:
Again, confusing. What's going on here--or rather, why? Renaming those `value.x` and `value.y` columns might help clarity.